"panic: send on closed channel" on amqp confirms

I'm hitting these errors from time to time:

```
panic: send on closed channel

goroutine 19094 [running]:
github.com/streadway/amqp.(*confirms).confirm(0xc420140000, 0x1, 0xc420140001)
        /home/dmishin/go/src/github.com/streadway/amqp/confirms.go:45 +0xac
github.com/streadway/amqp.(*confirms).One(0xc420140000, 0x1, 0x1001)
        /home/dmishin/go/src/github.com/streadway/amqp/confirms.go:66 +0x8c
github.com/streadway/amqp.(*Channel).dispatch(0xc42021e000, 0xe2ec00, 0xc420b42950)
        /home/dmishin/go/src/github.com/streadway/amqp/channel.go:309 +0x88d
github.com/streadway/amqp.(*Channel).recvMethod(0xc42021e000, 0xe2adc0, 0xc42022e500, 0xc42007a3d8, 0x1)
        /home/dmishin/go/src/github.com/streadway/amqp/channel.go:346 +0x34c
github.com/streadway/amqp.(*Connection).dispatchN(0xc4205d0140, 0xe2adc0, 0xc42022e500)
        /home/dmishin/go/src/github.com/streadway/amqp/connection.go:470 +0xd0
github.com/streadway/amqp.(*Connection).demux(0xc4205d0140, 0xe2adc0, 0xc42022e500)
        /home/dmishin/go/src/github.com/streadway/amqp/connection.go:429 +0x89
github.com/streadway/amqp.(*Connection).reader(0xc4205d0140, 0x7f9c40ee7200, 0xc42016a008)
        /home/dmishin/go/src/github.com/streadway/amqp/connection.go:521 +0x117
created by github.com/streadway/amqp.Open
        /home/dmishin/go/src/github.com/streadway/amqp/connection.go:228 +0x32d
```

Can't reproduce this in a test case. I'm using the pubsub example in my code. Could somebody help me debug the issue?

--- a/_examples/pubsub/pubsub.go
+++ b/_examples/pubsub/pubsub.go
@@ -87,14 +87,14 @@ func redial(ctx context.Context, url string) chan chan session {
 // publish publishes messages to a reconnecting session to a fanout exchange.
 // It receives from the application specific source of messages.
 func publish(sessions chan chan session, messages <-chan message) {
-	var (
-		running bool
-		reading = messages
-		pending = make(chan message, 1)
-		confirm = make(chan amqp.Confirmation, 1)
-	)
-
 	for session := range sessions {
+		var (
+			running bool
+			reading = messages
+			pending = make(chan message, 1)
+			confirm = make(chan amqp.Confirmation, 1)
+		)
+
 		pub := <-session
 
 		// publisher confirms for this channel/connection

